### ISSUE TITLE
Make "Owned By" section items clickable like "Frequent Users" entries

### ISSUE THREAD

**Issue by Mikhail-Ivanov** [lyft/amundsen#146]
Table documentation contains two sections with user emails: OWNED BY and FREQUENT USERS

![image](https://user-images.githubusercontent.com/2134660/67397591-70668580-f5ba-11e9-9738-49b35a41be99.png)

First one represents text as simple labels while second - as clickable links.
As a user I'd expect similar behavior for the both sections.

P.S. for my test application setup links lead to nowhere, maybe because of turned off user indexing.

**Comment by feng-tao** [comment]
cc @danwom let Daniel comment.

**Comment by danwom** [comment]
This is something that we've thought about internally as well. The `frequent users` are generated by ingesting usage data, so we know these entries exist as users in our graph. Table `owners` on the other hand, are largely populated manually with no validation. Sometimes the listed owners are teams, aliases, or even misspelled. These entries don't exist in our graph and would result in a 404 upon clicking.

We could validate the owners every time they are updated, but that's not without tradeoffs either. You might be sure the email you added was the right email, but you would be unable to add team emails, aliases, or anything that's not explicitly in the graph. Alternatively we could allow all owners to be added but then determine whether or not a corresponding user entry exists and link only those.

**Comment by Mikhail-Ivanov** [comment]
@danwom thank you for the explanation!

For now I've pushed PR without users validation: will check Amundsen user search API and think about verification approach a little bit later (initial impression - there is no strict email validation, similar emails considered as search hits).

**Comment by danwom** [comment]
@ttannis how do you feel about making table owners clickable without validation? I'm a bit hesitant since that might end up with a bunch of broken links that hurt trust.

**Comment by ttannis** [comment]
Couple thoughts:
1. I don't know if there would be broken links, so much as empty profile pages. I *believe* that since there is no validation, entering some incorrect owner email actually create a user node for them...so they would then end up having a profile page.
2. I'm not sure there is a quick fix here, historically a lot of the features around ownership were workaround after workaround -- all stemming from the fact that we do not have our ideal version of owner editing that requires all valid "People" entities (users + teams) to be indexed so that we can validate adding owners (metadata service would do this). 

In summary, if there is something we can do to improve the experience of broken links then perhaps let's move forward with that? But the next improvement we make regarding owners really should be a dedicated effort to do it right, because as we've seen building on top of what we have now isn't the best.

**Comment by feng-tao** [comment]
I am +1 on moving forward for the pr as well.

**Comment by Mikhail-Ivanov** [comment]
@ttannis accidental creation of new user nodes sounds like an harmful side effect...

@feng-tao in that case a new issue item regarding owners validation can be populated. I'm going to look into it but haven't had the opportunity yet

**Comment by Mikhail-Ivanov** [comment]
After some time spent on investigations I didn't find a handy approach for owners validation, checking whether the owner is a user or not (for instance creating a new table owner from modal dialog results into a new Neo4j node of User type).

If we had some additional field to distinguish users and owners at DB layer we could add it to User and PeopleUser interfaces. Then it would be fetched into the GlobalState through the function getTableOwnersFromResponseData and set to OwnerEditorProps at mapStateToProps. And finally it could be used during OwnerEditor component rendering to choose whether to add user profile link or not.

As an alternative we could add is_active field to User interface and show profile links only for "active" owners. Owners created through the modal dialog has this field set to false.
